pkg/pipeline/conntrack/hash.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package conntrack
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"hash/fnv"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+)
+
+// ComputeHash computes the hash of a flow log according to keyFields.
+// Two flow logs will have the same hash if they belong to the same connection.
+func ComputeHash(flowLog config.GenericMap, keyFields api.KeyFields) ([]byte, error) {
+	type hashType []byte
+	fieldGroup2hash := make(map[string]hashType)
+
+	// Compute the hash of each field group
+	for _, fg := range keyFields.FieldGroups {
+		h, err := computeHashFields(flowLog, fg.Fields)
+		if err != nil {
+			return nil, fmt.Errorf("compute hash: %w", err)
+		}
+		fieldGroup2hash[fg.Name] = h
+	}
+
+	// Compute the total hash
+	hash := fnv.New32a()
```

Comment:
Can we "trick" somehow (maybe interface?) so that it will be a parameter to the functions what length of Hash to use >??? 

@mariomac maybe you have an idea ??? ^^^

Reply:
Yeah, actually the `fnv.New32a` and rest of hashers implement the `io.Writer` interface. You can change the function signature to something like:

```go
func computeHashFields(flowLog config.GenericMap, fieldNames []string, hasher io.Writer) ([]byte, error) {
```

and invoke it like:

```go
h, err := computeHashFields(flowLog, fg.Fields, fnv.New32a())
```

Reply:
👍